src/nn/graph.ts

```diff
@@ -151,84 +127,95 @@ export class MLGraph {
   private operandRefs_: Map<MLOperand, number> = new Map();
   private constantTensors_: Map<ConstantOperand, tf.Tensor> = new Map();
 
-  compute(inputs: MLNamedInputs, outputs: MLNamedOutputs): void {
-    this.validateInputs(inputs);
-
-    // Validate and filter the required output operands.
-    utils.assert(Object.keys(outputs).length !== 0, 'The outputs is invalid.');
-    const outputOperands: Map<string, OutputOperand> = new Map();
-    for (const outputName in outputs) {
-      utils.assert(
-          typeof outputName === 'string' && this.outputs_.has(outputName),
-          'The name of the output is invalid.');
-      utils.assert(
-          utils.isTypedArray(outputs[outputName]),
-          'Only output of ArrayBufferView type is supported.');
-      outputOperands.set(outputName, this.outputs_.get(outputName));
+  private computeOutputTensors(
+    inputs: MLNamedArrayBufferViews = undefined,
+    outputs: MLNamedArrayBufferViews = undefined): tf.TensorContainerObject {
+    if (inputs) {
+      this.validateInputs(inputs);
+    } else {
+      inputs = {};
+      for (const inputName of this.inputs_.keys()) {
+        const inputOperand = this.inputs_.get(inputName);
+        const typedArrayConstructor =
+            utils.getTypedArray(inputOperand.desc.type);
+        const inputBuffer = new typedArrayConstructor(
+            utils.sizeFromDimensions(inputOperand.desc.dimensions));
+        inputs[inputName] = inputBuffer;
+      }
+    }
+    let outputOperands: Map<string, OutputOperand> = this.outputs_;
+    if (outputs) {
+      // Validate and filter the required output operands.
```

Comment:
Does it make sense to encapsulate the following code into `validateOutputs()` similar to `valdiateInputs()`?

Reply:
These following codes have two functions here, one is to validate `outputs` and the other is to set `outputOperands`  variable in the same one `for` loop, for executing efficiency, I don't plan to encapsulate them, is it OK for you? Thanks.

Reply:
Then please encapsulate the two functions into `validateAndSetOutputOperands()` returning the outputOperands.

Reply:
Thanks for your suggestion, I've updated the PR, please have another look, thanks.